app/controllers/repo_controller.rb

```diff
@@ -5,7 +5,31 @@ class RepoController < ApplicationController
 
   def index
     projects_response = GitlabRepoService.new(@current_user).fetch_user_projects(params[:page])
-    @projects = JSON.parse(projects_response).select { |project| project['visibility'] == 'public' }
+    projects = JSON.parse(projects_response).select { |project| project['visibility'] == 'public' }
```

Comment:
Stick to one action per method
https://en.wikipedia.org/wiki/Single-responsibility_principle